osu.Game.Rulesets.Catch/Difficulty/Skills/Movement.cs

```diff
@@ -25,9 +27,20 @@ public class Movement : Skill
         private float lastDistanceMoved;
         private double lastStrainTime;
 
-        public Movement(float halfCatcherWidth)
+        /// <summary>
+        /// The speed multiplier applied to the player's catcher.
+        /// </summary>
+        private readonly double catcherSpeedMultiplier;
+
+        public Movement(Mod[] mods, float halfCatcherWidth)
+            : base(mods)
         {
             HalfCatcherWidth = halfCatcherWidth;
+
+            // In catch, rate adjustment mods do not only affect the timings of hitobjects,
+            // but also the speed of the player's catcher, which has an impact on difficulty
+            var rateAdjustMod = mods.FirstOrDefault(m => m is ModRateAdjust);
+            catcherSpeedMultiplier = (rateAdjustMod as ModRateAdjust)?.SpeedChange.Value ?? 1;
```

Comment:
This is assuming there can only be a single `ModRateAdjust` applied at any given time.
This is the case right?
Would it be better to instead foolproof this by doing something like this instead?
```suggestion
            catcherSpeedMultiplier = 1;
            foreach (var rateAdjustMod in mods.Where(m => m is ModRateAdjust))
            {
                catcherSpeedMultiplier *= (rateAdjustMod as ModRateAdjust)?.SpeedChange.Value ?? 1;
            }
```

Reply:
`ModRateAdjust` is not the only mod that changes track rate. In fact, `IApplicableToRate` should be used for this. It was intended for such usages.

That might prove problematic depending on the overall structure, though, since `ModRateAdjust` assumes the rate is not constant, but a function of the track time instead (see `ModTimeRamp`). I don't know whether this code can handle that change of assumptions. If not, it's probably best to leave this be with a TODO.

Reply:
Ah I see, that is indeed tricky.
For `ModTimeRamp` to be properly accounted for, perhaps it makes more sense for `catcherSpeedMultiplier` to be set in the `DifficultyHitObject`...

Hmmm, I almost feel like there should be a more extensible way of describing the context for difficulty calculation.
The current system of passing a static clockrate to the `calculate()` function simply can't work with `ModTimeRamp`.
I'm happy to start putting together a solution for this variable clockrate issue, but I feel its beyond the scope of this PR.

As it is, this change is equivalent, so we can add a TODO to work on this further like you said.
However given that the `catcherSpeedMultiplier` is going to most likely need to exist at a `DifficutlyHitObject` level when we work on supporting variable clockrate, perhaps it would be better to simply close this PR and open a new one that merely renames the `ClockRate` field to `CatcherSpeedMultiplier` to prepare for the future changes.

Thoughts?

Reply:
I think the only way this can work is to support mods on a per-mod basis like is being done here, for edge cases like this. As in, the way it is in the PR seems better than using the `IApplicableToRate` interface for now.

That said, I'm not sure how important it is to be supporting all these new mods in difficulty calculation at this stage in development. Supporting just the constant rate adjusts (as you have it) seems fine for the time being. Adding a TODO as suggested is also recommended.

Shall we try and get this merged since it looks like this was the main blocker here?

Reply:
TODO added in 94701b77cb261963a66106c0b4270a59de6dfc24